blocks/search.py

```diff
@@ -143,29 +141,18 @@ def compute_contexts(self, inputs):
 
         Returns
         -------
-        A {name: :class:`numpy.ndarray`} dictionary of contexts ordered
-        like `self.context_names`.
-
-        """
-        contexts = self.context_computer(*[inputs[var]
-                                           for var in self.inputs])
-        return OrderedDict(equizip(self.context_names, contexts))
-
-    def compute_initial_states(self, contexts):
-        """Computes initial states.
-
-        Parameters
-        ----------
-        contexts : dict
-            A {name: :class:`numpy.ndarray`} dictionary of contexts.
-
-        Returns
-        -------
-        A {name: :class:`numpy.ndarray`} dictionary of states ordered like
+        A tuple containing a {name: :class:`numpy.ndarray`} dictionary of
+        contexts ordered like `self.context_names` and a
+        {name: :class:`numpy.ndarray`} dictionary of states ordered like
         `self.state_names`.
 
         """
-        return self.initial_state_computer(*list(contexts.values()))
+        outputs = self.initial_state_and_context_computer(
+            *[inputs[var] for var in self.inputs])
+        contexts = OrderedDict((n, outputs.pop(n)) for n in self.context_names)
+        self.beam_size = outputs.pop('beam_size')
```

Comment:
Please do not use class attributes for returning results from a method. You can add `beam_size` to the return statement.